src/main/java/org/jabref/gui/shared/SharedDatabaseLoginDialogView.java

```diff
@@ -0,0 +1,320 @@
+import org.jabref.model.database.shared.DatabaseNotSupportedException;
+import org.jabref.preferences.JabRefPreferences;
+
+import com.airhacks.afterburner.views.ViewLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SharedDatabaseLoginDialogView extends BaseDialog<Void> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SharedDatabaseLoginDialogView.class);
+
+    @FXML private ComboBox<DBMSType> cmbDbType;
+    @FXML private TextField tbHost;
+    @FXML private TextField tbDb;
+    @FXML private TextField tbPort;
+    @FXML private TextField tbUser;
+    @FXML private PasswordField tbPwd;
+    @FXML private CheckBox chkRememberPassword;
+    @FXML private TextField tbFolder;
+    @FXML private Button btnBrowse;
+    @FXML private CheckBox chkAutosave;
+
+    @Inject private DialogService dialogService;
+
+    private final SharedDatabasePreferences prefs = new SharedDatabasePreferences();
+
+    private final ButtonType connect = new ButtonType("Connect", ButtonData.YES);
+    private final Button btnConnect;
+    private final JabRefFrame frame;
+
+    private DBMSConnectionProperties connectionProperties;
+    private SharedDatabaseLoginDialogViewModel viewModel;
+
+    public SharedDatabaseLoginDialogView(JabRefFrame frame) {
+        this.frame = frame;
+        this.setTitle(Localization.lang("Connect to shared database"));
+
+        this.getDialogPane().getButtonTypes().addAll(connect, ButtonType.CANCEL);
+        this.btnConnect = (Button) this.getDialogPane().lookupButton(connect);
+        btnConnect.setOnAction(this::openDatabase);
+
+        ViewLoader.view(this)
+                  .load()
+                  .setAsContent(this.getDialogPane());
+
+    }
+
+    private void openDatabase(ActionEvent e) {
+
+        try {
+            checkFields();
+
+            connectionProperties = new DBMSConnectionProperties();
+            connectionProperties.setType(cmbDbType.getSelectionModel().getSelectedItem());
+            connectionProperties.setHost(tbHost.getText());
+            connectionProperties.setPort(Integer.parseInt(tbPort.getText()));
+            connectionProperties.setDatabase(tbDb.getText());
+            connectionProperties.setUser(tbUser.getText());
+            connectionProperties.setPassword(tbPwd.getText());
+
+            openSharedDatabase();
+        } catch (JabRefException exception) {
+            dialogService.showErrorDialogAndWait(Localization.lang("Warning"), exception);
+
+        }
+    }
+
+    @FXML
+    private void initialize() {
+
+        viewModel = new SharedDatabaseLoginDialogViewModel();
+        cmbDbType.itemsProperty().bind(viewModel.dbmstypeProperty());
+        cmbDbType.getSelectionModel().select(0);
+        viewModel.selectedDbmstypeProperty().bind(cmbDbType.getSelectionModel().selectedItemProperty());
+
+        viewModel.selectedDbmstypeProperty().addListener((observable, oldValue, newValue) -> {
```

Comment:
`EasyBind.subscribe` is usually easier to read. Moreover, this logic should also be moved to the view model.